### DISCUSSION TITLE
Creating a Custom Datatype

### DISCUSSION THREAD

**Discussion by MathewSalazarTS** [pimcore/pimcore#17535] [Q&A / Support]
https://pimcore.com/docs/platform/Pimcore/Extending_Pimcore/Bundle_Developers_Guide/Adding_Object_Datatypes/

I am trying to create a custom data type (FYI: I am new to pimcore, and I need this for Job's task). I want to create a select data type where based off the selection a list of checkboxes appear and the user will dynamic choose the options based on what they need. 

for example:
select options: XS-XL or 0-14
checklist: XS,S,etc... OR 0,2,etc...

Now looking at this documentation, I have a few questions:
Is there information outside of the link I provided from the documentation? or Video?

Where do we initialize them on the template? I see that the core bundle adds them in the index.twig file 

Is my custom data type even possible?

**Comment by MathewSalazarTS** [comment]
Update:
_Where do we initialize them on the template? I see that the core bundle adds them in the index.twig file"_

I noticed that I can add the JS scripts from my bundle's main php file.

`public function getJsPaths(): array`

**Reply by fashxp** [reply]
alternatively, you also can add them via event listener: https://pimcore.com/docs/platform/Pimcore/Extending_Pimcore/Bundle_Developers_Guide/Loading_Admin_UI_Assets

**Comment by fashxp** [comment]
Yes, your datatype is definitely possible to implement.
